src/main/scala/com/horizon/exchangeapi/OrgsRoutes.scala

```diff
@@ -598,32 +587,21 @@ trait OrgsRoutes extends JacksonSupport with AuthenticationSupport {
         complete({
           // Variables to help with building the query
           val lastTime = reqBody.lastUpdated.getOrElse(ApiTime.beginningUTC)
-          //perf: reduce these 2 db queries to 1 db query
-          var qOrgQuery = ResourceChangesTQ.rows.filter(_.orgId === orgId).filter(_.lastUpdated >= lastTime).filter(_.changeId >= reqBody.changeId)
+          // filter by lastUpdated and changeId then filter by either it's in the org OR it's not in the same org but is public
+          var qFilter = ResourceChangesTQ.rows.filter(_.lastUpdated >= lastTime).filter(_.changeId >= reqBody.changeId).filter(u => (u.orgId === orgId) || (u.orgId =!= orgId && u.public === "true"))
           ident match {
             case _: INode =>
-              qOrgQuery = qOrgQuery.filter(u => (u.category === "node" && u.id === ident.getIdentity) || u.category =!= "node")
+              // if its a node calling then it doesn't want information about any other nodes
+              qFilter = qFilter.filter(u => (u.category === "node" && u.id === ident.getIdentity) || u.category =!= "node")
             case _ => ;
           }
-          val qOrg = for { r <- qOrgQuery } yield r
-          val qPublic = for {
-            r <- ResourceChangesTQ.rows.filter(_.orgId =!= orgId).filter(_.public === "true").filter(_.lastUpdated >= lastTime).filter(_.changeId >= reqBody.changeId)
-          } yield r
-
-          var qOrgResp : scala.Seq[ResourceChangeRow] = null
-          var qPublicResp : scala.Seq[ResourceChangeRow] = null
-
-          db.run(qOrg.result.asTry.flatMap({
-            case Success(qOrgResult) =>
-              //logger.debug("POST /orgs/" + orgId + "/changes changes in caller org: " + qOrgResult.toString())
-              logger.debug("POST /orgs/" + orgId + "/changes changes in caller org: " + qOrgResult.size)
-              qOrgResp = qOrgResult
-              qPublic.result.asTry
-            case Failure(t) => DBIO.failed(t).asTry
-          }).flatMap({
-            case Success(qIBMResult) => qPublicResp = qIBMResult
-              //logger.debug("POST /orgs/" + orgId + "/changes public changes in IBM org: " + qIBMResult.toString())
-              logger.debug("POST /orgs/" + orgId + "/changes public changes in IBM org: " + qIBMResult.size)
+          val q = for { r <- qFilter.sortBy(_.changeId) } yield r //sort the response by changeId
```

Comment:
I realized that now that it is a single query in changeId order that you can limit the query itself to `maxRecords` using:
```
qFilter.sortBy(_.changeId).take(maxRecords)
```

In the case, for example, when the node's network connection to the exchange was down for a while, this would prevent the query from getting a lot of rows that you end of ignoring in the subsequent processing loop.